.github/actions/install-winget/action.yaml

```diff
@@ -1,78 +1,52 @@
-
-        # Detect runner architecture
-        $Architecture = if ($env:RUNNER_ARCH -eq "ARM64") { "arm64" } else { "x64" }
-        Write-Host "Runner architecture: $Architecture"
-
-        # winget-cli release v1.11.400
-        # Define the winget-cli release tag to use
-        $WingetReleaseTag = "v1.11.400"
-        # Fetch release info by tag instead of magic number release ID
-        $ReleaseInfo = Invoke-RestMethod -Headers $AuthenticatedHeaders "https://api.github.com/repos/microsoft/winget-cli/releases/tags/$WingetReleaseTag"
-        $WingetDownloadUri = $ReleaseInfo.assets.browser_download_url | Where-Object { $_.EndsWith('.msixbundle') }
-        $WingetLicenseDownloadUri = $ReleaseInfo.assets.browser_download_url | Where-Object { $_.EndsWith('License1.xml') }
-        $WingetDependenciesZipDownloadUri = $ReleaseInfo.assets.browser_download_url | Where-Object { $_.EndsWith('DesktopAppInstaller_Dependencies.zip') }
-
-        # Print to logs
-        Write-Host "WingetDownloadUri=$WingetDownloadUri"
-        Write-Host "WingetLicenseDownloadUri=$WingetLicenseDownloadUri"
-        Write-Host "WingetDependenciesZipDownloadUri=$WingetDependenciesZipDownloadUri"
-
-        # Save output for next step
-        Write-Output "WingetDownloadUri=$WingetDownloadUri" >> $env:GITHUB_ENV
-        Write-Output "WingetLicenseDownloadUri=$WingetLicenseDownloadUri" >> $env:GITHUB_ENV
-        Write-Output "Architecture=$Architecture" >> $env:GITHUB_ENV
-        Write-Output "WingetDependenciesZipDownloadUri=$WingetDependenciesZipDownloadUri" >> $env:GITHUB_ENV
-        Write-Output "InstallWingetTempDir=$env:RUNNER_TEMP/install-winget" >> $env:GITHUB_ENV
-
-    - name: Download Winget Assets and Dependencies
-      shell: pwsh
-      run: |
-        $AuthenticatedHeaders = @{ "Authorization" = "Bearer ${{ inputs.GITHUB_TOKEN }}" }
-        New-Item -Type Directory $env:InstallWingetTempDir
-        
-        # Download winget and license (architecture-agnostic)
-        Invoke-WebRequest -Headers $AuthenticatedHeaders -Uri $env:WingetDownloadUri -OutFile $env:InstallWingetTempDir/winget.msixbundle
-        Invoke-WebRequest -Headers $AuthenticatedHeaders -Uri $env:WingetLicenseDownloadUri -OutFile $env:InstallWingetTempDir/license.xml
-        Invoke-WebRequest -Headers $AuthenticatedHeaders -Uri $env:WingetDependenciesZipDownloadUri -OutFile $env:InstallWingetTempDir/DesktopAppInstaller_Dependencies.zip
+        $raw = "${{ inputs.force-cli-install }}"
+
+        $normalizedValue = $raw.Trim().ToLowerInvariant()
+
+        switch ($normalizedValue) {
+          'true' {
+            $normalized = 'true'
+            break
+          }
+          'false' {
+            $normalized = 'false'
+            break
+          }
+          default {
+            Write-Error "Invalid force-cli-install value '$raw'. Acceptable values: true or false."
+            exit 1
+          }
+        }
```

Comment:
This can be simplified to:

```pwsh
$raw = "${{ inputs.force-cli-install }}"
$normalized = $raw.Trim().ToLowerInvariant()
if ($normalized -ne 'true' -and $normalized -ne 'false') {
    Write-Error "Invalid force-cli-install value '$raw'. Acceptable values: true or false."
    exit 1
}
```